### PR TITLE
Create Redis staging setup

### PR DESCRIPTION
**Pull request by KyleJu**
Taken from #2146. Create Redis staging setup using the [set-up instructions](https://docs.google.com/document/d/1hV5MTrGjtKy5kyyk2983kJz3POUWSOBbH-t-qBRQ7Ns/edit?usp=sharing&resourcekey=0-hdThpHi-WIM-48sl4EfO6Q).

- Create app.staging.yaml for Redis staging setup. All non-Redis fields remain the same
- Set 4GB cache limit on Redis staging
- Tested in staging in #2146

Note that there are separate staging and production Redis setup

### CODE REVIEW

File: scripts/deploy_site.sh [hunk 1]
```diff
@@ -29,6 +29,6 @@ gcloud beta app deploy \
   --version $deployVersion \
   --no-promote \
   $BASEDIR/../notifier.yaml \
-  $BASEDIR/../app.yaml  \
+  $BASEDIR/../app.staging.yaml  \
```
Comment (jrobbins) [inline scripts/deploy_site.sh:32]: With this change, how would we deploy to prod?
I think we need a conditional in here somewhere, or we need to specify a command-line argument with the name of the yaml file to deploy (or just the 'staging' or 'prod' part of it).

**Review by jrobbins — CHANGES_REQUESTED** [review]

**Review by jrobbins — APPROVED** [review]

**Comment by jrobbins** [comment]
It looks like you added a ton of files to this PR.

**Comment by KyleJu** [comment]
> It looks like you added a ton of files to this PR.

I merged two other PRs into this PR. Should I have done the other way around? (Merge to main one at a time)

**Comment by KyleJu** [comment]
The files being merged are reviewed in https://github.com/GoogleChrome/chromium-dashboard/pull/2212. Everything looks good so I will merge this PR to main

**Comment by jrobbins** [comment]
Ah, OK.
I think the best way to do it would have been:
1. Merge your other PRs into main on GitHub.com
2. git checkout main
3. git pull
4. git checkout redis-setup
5. git merge main
6. If there are any conflicts resolve them and commit, but in this case I don't think there would be any
7. git push

**Comment by KyleJu** [comment]
> Ah, OK. I think the best way to do it would have been:
> 
> 1. Merge your other PRs into main on GitHub.com
> 2. git checkout main
> 3. git pull
> 4. git checkout redis-setup
> 5. git merge main
> 6. If there are any conflicts resolve them and commit, but in this case I don't think there would be any
> 7. git push

Since the three PRs are parent/child, we can probably execute your command from the parent PR to children PRs, but yea merging to main one at a time is a better option!

**Pull request merged**